### ISSUE TITLE
Unconditional commands with loops

### ISSUE THREAD

**Issue by jcassee** [willthames/ansible-lint#281]
# Issue Type
- Bug report

# Ansible and Ansible Lint details

```
$ ansible --version
ansible 2.3.2.0
  config file = /home/joost/Ontwikkeling/GoAbout/goabout-ansible2/ansible.cfg
  configured module search path = [u'library']
  python version = 2.7.12 (default, Nov 19 2016, 06:48:10) [GCC 5.4.0 20160609]

$ ansible-lint --version
ansible-lint 3.4.15
```

- ansible installation method: one of source, pip, OS package
pip

- ansible-lint installation method: one of source, pip, OS package
pip

# Desired Behaviour

This playbook should not produce a warning:
```
- hosts: all
  tasks:
    - name: Find old source files
      find:
        paths: /tmp/source
        age: 2d
      register: find

    - name: Archive old source files
      command: mv {{ item }} /tmp/archive
      with_items: "{{ find.files }}"
```

# Actual Behaviour (Bug report only)

```
$ ansible-lint test.yml
[ANSIBLE0012] Commands should not change things if nothing needs doing
test.yml:9
Task/Handler: Archive old source files
````

**Comment by willthames** [comment]
It should warn, as none of the protections in 
http://willthames.github.io/2016/09/21/using-command-and-shell-in-ansible.html
are used.

Even if you added

```
changed_when: find.files
```

that would meet the criteria (it doesn't do anything if no files are found, so changed will report as false)

**Comment by jcassee** [comment]
Well, if `find.files` is empty no task will be executed, so no warning is generated:

```
[...]
TASK [Find old source files] *****************************************************************************************************
ok: [localhost]

TASK [Archive old source files] **************************************************************************************************

PLAY RECAP ***********************************************************************************************************************
localhost               : ok=2    changed=0    unreachable=0    failed=0   
```

But whether you feel this is a potential convergence problem might be a matter of taste.
